File: API_FLASK/Model/Banco.py
import mysql.connector
from mysql.connector import Error

class Banco:
    HOST = '127.0.0.1'
    USER = 'root'
    PWD = ''
    DB = 'api_rest'
    PORT = 3306
    CONEXAO = None
    
    @staticmethod
    def conectar():
        # Verifica se já existe uma conexão estabelecida
        if Banco.CONEXAO is None:
            try:
                # Tenta estabelecer uma nova conexão utilizando as informações fornecidas
                Banco.CONEXAO = mysql.connector.connect(
                    host=Banco.HOST,
                    user=Banco.USER,
                    password=Banco.PWD,
                    database=Banco.DB,
                    port=Banco.PORT
                )
                print("Conexão com o banco de dados estabelecida")
            except Error as err:
                obj_resposta = {
                    'cod': 1,
                    'msg': "Erro ao conectar no banco",
                    'erro': str(err)
                }
                print(obj_resposta)  # Log de erro
                exit(1)  # Encerra o script em caso de erro

    # Método público para obter a conexão com o banco de dados
    @staticmethod
    def getConexao():
        # Verifica se já existe uma conexão estabelecida
        if Banco.CONEXAO is None:
            # Se não houver, estabelece uma nova conexão
            Banco.conectar()
        # Retorna a conexão
        return Banco.CONEXAO

File: model/Banco.js
const mysql = require('mysql2/promise');

class Banco {
    static host = 'localhost';
    static user = 'root';
    static password = '';
    static banco = 'api_rest';
    static porta = 3306;
    static conexao = null;

    static async conectarBanco() {
        if (Banco.conexao === null) {
            try {
                Banco.conexao = await mysql.createConnection({
                    host: Banco.host,
                    user: Banco.user,
                    password: Banco.password,
                    database: Banco.banco,
                    port: Banco.porta
                });
                console.log("Conexão bem-sucedida!");
            } catch (error) {
                let resposta = {
                    cod: 1,
                    erro: error.message
                };
                console.error(JSON.stringify(resposta)); // Exibe o JSON de erro
                throw error;
            }
        }
    }

    static async getConexao() {
        if (Banco.conexao === null) {
            await Banco.conectarBanco();
        }
        return Banco.conexao;
    }
}

module.exports = Banco;
